azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -36,48 +37,131 @@ class CGroupConfigurator(object):
+
+                if memory_controller_root is not None:
+                    logger.info("The memory cgroup controller is mounted at {0}", memory_controller_root)
+                else:
+                    log_cgroup_warn("The memory cgroup controller is not mounted")
+
+                #
+                # check v2 controllers
+                #
+                cgroup2_mountpoint, cgroup2_controllers = self._cgroups_api.get_cgroup2_controllers()
+                if cgroup2_mountpoint is not None:
+                    log_cgroup_warn("cgroups v2 mounted at {0}.  Controllers: [{1}]", cgroup2_mountpoint, cgroup2_controllers)
+
+                #
+                # check the cgroups for the agent
+                #
+                agent_unit_name = get_osutil().get_service_name() + ".service"
+                cpu_cgroup_relative_path, memory_cgroup_relative_path = self._cgroups_api.get_cpu_and_memory_cgroup_relative_paths_for_process("self")
+                if cpu_cgroup_relative_path is None:
+                    log_cgroup_warn("The agent's process is not within a CPU cgroup")
+                else:
+                    cpu_accounting = self._cgroups_api.get_unit_property(agent_unit_name, "CPUAccounting")
+                    log_cgroup_info('CPUAccounting: {0}', cpu_accounting)
+
+                if memory_cgroup_relative_path is None:
+                    log_cgroup_warn("The agent's process is not within a memory cgroup")
+                else:
+                    memory_accounting = self._cgroups_api.get_unit_property(agent_unit_name, "MemoryAccounting")
+                    log_cgroup_info('MemoryAccounting: {0}', memory_accounting)
+
+                #
+                # All good, enable cgroups and start monitoring the agent
+                #
+                self._cgroups_enabled = True
+
+                if cpu_controller_root is None or cpu_cgroup_relative_path is None:
+                    logger.info("Will not track CPU for the agent's cgroup")
+                else:
+                    cpu_cgroup_path = os.path.join(cpu_controller_root, cpu_cgroup_relative_path)
+                    CGroupsTelemetry.track_cgroup(CpuCgroup(agent_unit_name, cpu_cgroup_path))
+
+                if memory_controller_root is None or memory_cgroup_relative_path is None:
+                    logger.info("Will not track memory for the agent's cgroup")
+                else:
+                    memory_cgroup_path = os.path.join(memory_controller_root, memory_cgroup_relative_path)
+                    CGroupsTelemetry.track_cgroup(MemoryCgroup(agent_unit_name, memory_cgroup_path))
 
-            add_event(
-                AGENT_NAME,
-                version=CURRENT_VERSION,
-                op=WALAEventOperation.InitializeCGroups,
-                is_success=self._enabled,
-                message=status,
-                log_event=False)
+            except Exception as e:
+                message = "Error initializing cgroups: {0}".format(ustr(e))
```

Comment:
Not sure if needed, but shouldn't we mark `self._cgroups_enabled = False` here?

Reply:
initially I was setting enabled to True at the very end of the try clause, but then I realized cgroups should be ready to use if we reach line 133 so I moved the assignment there. If for any reason we cannot track the agent cgroups we should still track extensions; if for any reason we cannot track the agent's memory cgroup we should still track its cpu cgroup